app/actions/CisUserDataFilterAction.scala

```diff
@@ -34,6 +34,7 @@ case class CisUserDataFilterAction(taxYear: Int,
   override protected[actions] def executionContext: ExecutionContext = ec
 
   override protected[actions] def filter[A](input: UserSessionDataRequest[A]): Future[Option[Result]] = {
+    // TODO: the compiler complains this match may not be exhaustive - refactor
     val result = input.cisUserData match {
```

Comment:
Since you have added a TODO, you can add @unchecked so that the warning will disappear.